sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -244,37 +240,32 @@ export class AppConfigurationClient {
     id: ConfigurationSettingId,
     options: GetConfigurationSettingOptions = {}
   ): Promise<GetConfigurationSettingResponse> {
-    const requestOptions = operationOptionsToRequestOptionsBase(options);
-    return await this.spanner.trace(
-      "getConfigurationSetting",
-      requestOptions,
-      async (newOptions) => {
-        const originalResponse = await this.client.getKeyValue(id.key, {
-          ...newOptions,
-          label: id.label,
-          select: formatFieldsForSelect(options.fields),
-          ...formatAcceptDateTime(options),
-          ...checkAndFormatIfAndIfNoneMatch(id, options)
-        });
-
-        const response: GetConfigurationSettingResponse = transformKeyValueResponseWithStatusCode(
-          originalResponse
-        );
+    return await this._trace("getConfigurationSetting", options, async (newOptions) => {
```

Comment:
@chradek , as co-author of the appconfig package - it appears our tracing has been broken/incorrect for a bit. So this also fixes it and adds in some tests as well.